lib/utils/autocomplete.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const homedir = require('os').homedir();
+const readFileSync = require('./fs/readFileSync');
+const fileExistsSync = require('./fs/fileExistsSync');
+
+const tab = require('tabtab')({
+  name: 'serverless',
+});
+
+const autocomplete = () => {
+  const cacheFilePath = `${homedir}/.serverless/autocomplete.json`;
+  if (!fileExistsSync(cacheFilePath)) {
+    return;
+  }
+  const cacheFile = readFileSync(cacheFilePath);
+
+  tab.on('serverless', (data, done) => {
+    done(null, Object.keys(cacheFile));
```

Comment:
I would nest the location of the cache within `autocomplete.json` to a sub property. This will leave some room for any future changes that may be necessary to the autocomplete structure and allow for modifications in future versions/expansion by plugins.